### ISSUE TITLE
ENOENT, insufficient config validating

### ISSUE THREAD

**Issue by devnull-ed** [winstonjs/winston-daily-rotate-file#129]
Hello.
There were few issues related to ENOENT error. At this time, winston fails with enoent on unproperly formatted datePattern.
Some chars are not valid for file naming in some file systems. For example I get enoent error when I set datePattern with colon in it.

```
YYYY-MM-DD_HH:mm:ss
```

Results in:

```
Debugger attached.
events.js:182
      throw er; // Unhandled 'error' event
      ^

Error: ENOENT: no such file or directory, open 'C:\Users\User\Projects\my-project\log.2018-04-02_17:57:50'


Waiting for the debugger to disconnect...
```

That's obivious why winston fails. UNIX systems allow colon in file naming while Windows systems dont.

https://stackoverflow.com/a/22988785

**Comment by mattberther** [comment]
Resolved with #136 and pushed to npm as `winston-daily-rotate-file@3.1.0`

**Comment by devnull-ed** [comment]
Sweet
